src/vmm/src/vmm_config/memory.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fmt;
+use std::sync::{Arc, Mutex};
+
+use devices::virtio::Memory;
+use serde::{Deserialize, Serialize};
+
+const MIB: u64 = 1024 * 1024;
+
+type MutexMemory = Arc<Mutex<Memory>>;
+
+/// Errors associated with the operations allowed on the memory.
+#[derive(Debug, derive_more::From)]
+pub enum MemoryConfigError {
+    /// The user made a request on an inexistent memory device.
+    DeviceNotFound,
+    /// Device not activated yet.
+    DeviceNotActive,
+    /// There already exists a device with this id.
+    DeviceWithThisIdExists,
+    /// Failed to create a memory device.
+    CreateFailure(devices::virtio::memory::Error),
+}
+
+impl fmt::Display for MemoryConfigError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
+        use self::MemoryConfigError::*;
+        match self {
+            DeviceNotFound => write!(f, "No memory device found."),
+            DeviceNotActive => write!(
+                f,
+                "Device is inactive, check if memory driver is enabled in guest kernel."
+            ),
+            DeviceWithThisIdExists => write!(f, "A memory device with this id already exists"),
+            CreateFailure(err) => write!(f, "Error creating the memory device: {:?}", err),
```

Comment:
Might as well implement `Display` for `memory::Error` too and pretty-print all the way :)

Reply:
Done